modules/adagioRtdProvider.js

```diff
@@ -0,0 +1,680 @@
+
+        data.session = {
+          rnd,
+          new: isNew || false, // legacy: `new` was used but the choosen name is not good.
+          // Don't use values if they are not defined.
+          ...(vwSmplg !== undefined && { vwSmplg }),
+          ...(vwSmplgNxt !== undefined && { vwSmplgNxt }),
+          ...(lastActivityTime !== undefined && { lastActivityTime })
+        }
+
+        if (isNewSession(lastActivityTime)) {
+          data.session.new = true;
+          data.session.rnd = Math.random();
+        }
+
+        _internal.getAdagioNs().queue.push({
+          action: 'session',
+          ts: Date.now(),
+          data: {
+            session: {
+              ...data.session
+            }
+          }
+        });
+      });
+    },
+    get: function() {
+      return data.session;
+    }
+  }
+})();
+
+const _FEATURES = (function() {
+  /**
+   * @type {Features}
+   */
+  const features = {
+    initialized: false,
+    data: {},
+  };
+
+  return {
+    // reset is used for testing purpose
+    reset: function() {
+      features.initialized = false;
+      features.data = {};
+    },
+    get: function() {
+      if (!features.initialized) {
+        features.data = {
+          page_dimensions: getPageDimensions().toString(),
+          viewport_dimensions: getViewPortDimensions().toString(),
+          user_timestamp: getTimestampUTC().toString(),
+          dom_loading: getDomLoadingDuration().toString(),
+        }
+        features.initialized = true;
+      }
+
+      return { ...features.data };
+    }
+  }
+})();
+
+export const _internal = {
+  getAdagioNs: function() {
+    return _ADAGIO;
+  },
+
+  getSession: function() {
+    return _SESSION;
+  },
+
+  getFeatures: function() {
+    return _FEATURES;
+  },
+
+  /**
+   * Ensure that the bidder is Adagio.
+   *
+   * @param {string} alias
+   * @returns {boolean}
+   */
+  isAdagioBidder: function (alias) {
+    if (!alias) {
+      return false
+    }
+    return (alias + adapterManager.aliasRegistry[alias]).toLowerCase().includes(ADAGIO_BIDDER_CODE);
+  },
+
+  /**
+   * Returns the session data from the localStorage.
+   *
+   * @param {string} storageValue - The value stored in the localStorage.
+   * @returns {Session}
+   */
+  getSessionFromLocalStorage: function(storageValue) {
+    const _default = {
+      new: true,
+      rnd: Math.random()
+    };
+
+    const obj = JSON.parse(storageValue, function(name, value) {
+      if (name.charAt(0) !== '_' || name === '') {
+        return value;
+      }
+    });
+
+    return (!obj || !obj.session) ? _default : obj.session;
+  }
+}
+
+function loadAdagioScript(config) {
+  storage.localStorageIsEnabled(isValid => {
+    if (!isValid) {
+      return
+    }
+
+    loadExternalScript(SCRIPT_URL, SUBMODULE_NAME, undefined, undefined, { id: `adagiojs-${getUniqueIdentifierStr()}`, 'data-pid': config.params.organizationId });
```

Comment:
@osazos what is the script doing at this point? Why not just put all the source of the script in your bid adapter? If you truly need script that only advantages your bidder, your only option is to convince publishers to source it. RTD modules must not favor a single bidder.